app/src/main/res/layout/widget_media_controls.xml

```diff
@@ -0,0 +1,135 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/widgetLayout"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:minHeight="40dp"
+    android:minWidth="40dp"
+    android:background="@drawable/widget_button_background">
+
+    <LinearLayout
+        android:id="@+id/widgetImageButtonLayout"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="horizontal">
+
+        <FrameLayout
+            android:layout_width="60dp"
+            android:layout_height="match_parent">
+            <ImageView
+                android:id="@+id/widgetMediaImage"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:visibility="invisible"
+                android:contentDescription="@string/widget_media_image_description"
+                android:scaleType="centerCrop" />
+            <ImageView
+                android:id="@+id/widgetMediaPlaceholder"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:layout_marginStart="4dp"
+                android:layout_marginTop="4dp"
+                android:layout_marginBottom="4dp"
+                android:src="@drawable/app_icon"
+                android:contentDescription="@string/widget_media_image_description"
+                android:scaleType="fitCenter" />
```

Comment:
Can we just use the same ImageView and replace the image when we have/don't have an image?

Reply:
The reason I have it set like this is because of the margins. If the image is supposed to take up the entire left-hand section, it can't have any margins on it, but the placeholder looks silly if it is touching the edge. I can give you some screenshots of what it looks like if it's just straight replacement if you'd like.